Sources/SwiftFusion/Inference/ChordalInitialization.swift

```diff
@@ -0,0 +1,252 @@
+// Copyright 2020 The SwiftFusion Authors. All Rights Reserved.
+//
+// Licensed under the Apache License, Version 2.0 (the "License");
+// you may not use this file except in compliance with the License.
+// You may obtain a copy of the License at
+//
+//     http://www.apache.org/licenses/LICENSE-2.0
+//
+// Unless required by applicable law or agreed to in writing, software
+// distributed under the License is distributed on an "AS IS" BASIS,
+// WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+// See the License for the specific language governing permissions and
+// limitations under the License.
+
+import PenguinStructures
+import TensorFlow
+
+/// A BetweenFactor alternative that uses the Chordal (Frobenious) norm on rotation for Rot3
+/// Please refer to Carlone15icra (Initialization Techniques for 3D SLAM: a Survey on Rotation Estimation and its Use in Pose Graph Optimization)
+/// for explanation.
+public struct RelaxedRotationFactorRot3: LinearizableFactor
+{
+  public typealias Variables = Tuple2<Matrix3, Matrix3>
+  public typealias JacobianRows = Array9<Tuple2<Matrix3.TangentVector, Matrix3.TangentVector>>
+  
+  public let edges: Variables.Indices
+  public let difference: Matrix3
+  
+  public init(_ id1: TypedID<Matrix3, Int>, _ id2: TypedID<Matrix3, Int>, _ difference: Matrix3) {
+    self.edges = Tuple2(id1, id2)
+    self.difference = difference
+  }
+  
+  public typealias ErrorVector = Matrix3
+  
+  @differentiable
+  public func errorVector(_ start: Matrix3, _ end: Matrix3) -> ErrorVector {
```

Comment:
can we not call it start and end, but R1 and R2?

Reply:
done